java/src/main/java/com/saucelabs/saucebindings/SauceOptions.java

```diff
@@ -326,7 +326,38 @@ private void setEnumCapability(String key, String value) {
         }
     }
 
-    protected String getEnvironmentVariable(String key) {
+    private MutableCapabilities addAuthentication() {
+        MutableCapabilities caps = new MutableCapabilities();
+        caps.setCapability("username", getSauceUsername());
+        caps.setCapability("accessKey", getSauceAccessKey());
+        return caps;
+    }
+
+    String getSauceUsername() {
+        if (getSystemProperty("SAUCE_USERNAME") != null) {
+            return getSystemProperty("SAUCE_USERNAME");
+        } else if (getEnvironmentVariable("SAUCE_USERNAME") != null) {
+            return getEnvironmentVariable("SAUCE_USERNAME");
+        } else {
+            throw new SauceEnvironmentVariablesNotSetException("Sauce Username was not provided");
+        }
+    }
+
+    String getSauceAccessKey() {
+        if (getSystemProperty("SAUCE_ACCESS_KEY") != null) {
+            return getSystemProperty("SAUCE_ACCESS_KEY");
+        } else if (getEnvironmentVariable("SAUCE_ACCESS_KEY") != null) {
+            return getEnvironmentVariable("SAUCE_ACCESS_KEY");
+        } else {
+            throw new SauceEnvironmentVariablesNotSetException("Sauce Access Key was not provided");
+        }
+    }
+
+    String getSystemProperty(String key) {
+        return System.getProperty(key);
+    }
+
+    String getEnvironmentVariable(String key) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=5305593)